cypress/integration/channel/message_spec.js

```diff
@@ -7,35 +7,67 @@
 // - Use element ID when selecting an element. Create one if none.
 // ***************************************************************
 
+/*eslint max-nested-callbacks: ["error", 3]*/
+
 describe('Message', () => {
     it('should show only 1 header when user send multiple posts', () => {
         // 1. Login as sysadmin and go to /
         cy.login('sysadmin');
         cy.visit('/');
 
         // 2. Post a message to force next user message to display a message
-        cy.get('#post_textbox').type('Hello').type('{enter}');
+        cy.postMessage('Hello');
 
         // 3. Login as "user-1" and go to /
         cy.login('user-1');
         cy.visit('/');
 
         // 4. Post message "One"
-        cy.get('#post_textbox').type('One').type('{enter}');
+        cy.postMessage('One');
 
         // * Check profile image is visible
         cy.get('#postListContent > .post.current--user > .post__content > .post__img').last().should('be.visible').find('span > img').should('be.visible');
 
         // 5. Post message "Two"
-        cy.get('#post_textbox').type('Two').type('{enter}');
+        cy.postMessage('Two');
 
         // * Check profile image is not visible
         cy.get('#postListContent > .post.current--user > .post__content > .post__img').last().should('be.visible').should('be.empty');
 
         // 6. Post message "Three"
-        cy.get('#post_textbox').type('Three').type('{enter}');
+        cy.postMessage('Three');
 
         // * Check profile image is not visible
         cy.get('#postListContent > .post.current--user > .post__content > .post__img').last().should('be.visible').should('be.empty');
     });
+
+    it('M14012 Focus move to main input box when a character key is selected', () => {
+        // 1. Login and go to /
+        cy.login('user-1');
+        cy.visit('/');
+
+        // 2. Post message
+        cy.postMessage('Message');
+
+        cy.getLastPostId().then((postId) => {
+            const divPostId = `#post_${postId}`;
+
+            // 3. Left click on post to move the focus out of the main input box
+            cy.get(divPostId).click();
+
+            // 4. Push a character key such as "A"
+            cy.get('#post_textbox').type('A');
+
+            // 5. Open the "..." menu on a post in the main to move the focus out of the main input box
+            cy.get(divPostId).trigger('mouseover');
+            cy.get(`#CENTER_dropdown_${postId} .dropdown-toggle`).click({force: true});
+        });
+
+        // 6. Push a character key such as "A"
+        cy.get('#post_textbox').type('A');
```

Comment:
Hey @courtneypattison Can you move this inside the `then` func above? That way it makes it obvious that these commands are to be followed by the execution of `promise` and not independent?

Reply:
@courtneypattison Let me know if you need help on this. Can't wait to see this merge in our codebase.
By the way, you did a good job in this PR!  I've got an idea on how to test other parts of the Post Header - https://github.com/mattermost/mattermost-webapp/pull/2389.

Reply:
Hey @sudheerDev Good idea! I think I made the change you requested. @saturninoabril  Thank you! I think it's done :)

Reply:
@courtneypattison One minor thing. There is an lint error https://build.mattermost.com/blue/organizations/jenkins/mw%2Fmattermost-webapp/detail/PR-2377/4/pipeline

`64:1  error  Trailing spaces not allowed  no-trailing-spaces` 
and if you can merge master into this branch just to get it upto date that will be great too. I will merge it as soon as these are addressed.

Reply:
@sudheerDev Sorry, that was sloppy! I've made the fix and merged, hopefully! My git skills are pretty weak

Reply:
Uh oh, I think I messed the merge up. Will try to fix!

Reply:
No problem. If you are not sure about how to resolve i suggest an easy path for now.

1. Create a new branch from this branch(For worst case scenario).
2. Copy all the commit Id's you need for the PR.
2. Go to master delete this branch locally 
3. Create new branch with the same name
4. Git cherrypick all the earlier commits you copied and force push.

Though this is not idle it will work for now

Reply:
Sorry about that! It should be fixed now.